presto-kudu/src/test/java/io/prestosql/plugin/kudu/TestKuduIntegrationDynamicFilter.java

```diff
@@ -0,0 +1,307 @@
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import io.prestosql.Session;
+import io.prestosql.execution.QueryStats;
+import io.prestosql.operator.OperatorStats;
+import io.prestosql.spi.QueryId;
+import io.prestosql.spi.connector.ColumnHandle;
+import io.prestosql.spi.connector.DynamicFilter;
+import io.prestosql.spi.predicate.TupleDomain;
+import io.prestosql.testing.AbstractTestQueryFramework;
+import io.prestosql.testing.DistributedQueryRunner;
+import io.prestosql.testing.MaterializedResult;
+import io.prestosql.testing.QueryRunner;
+import io.prestosql.testing.ResultWithQueryId;
+import io.prestosql.tpch.TpchTable;
+import org.intellij.lang.annotations.Language;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+import static com.google.common.collect.ImmutableSet.toImmutableSet;
+import static io.prestosql.SystemSessionProperties.ENABLE_LARGE_DYNAMIC_FILTERS;
+import static io.prestosql.SystemSessionProperties.JOIN_DISTRIBUTION_TYPE;
+import static io.prestosql.SystemSessionProperties.JOIN_REORDERING_STRATEGY;
+import static io.prestosql.plugin.kudu.KuduQueryRunnerFactory.createKuduQueryRunnerTpch;
+import static io.prestosql.plugin.kudu.KuduSplitManager.completedOrTimeout;
+import static io.prestosql.sql.analyzer.FeaturesConfig.JoinDistributionType.BROADCAST;
+import static io.prestosql.sql.analyzer.FeaturesConfig.JoinDistributionType.PARTITIONED;
+import static io.prestosql.sql.analyzer.FeaturesConfig.JoinReorderingStrategy.NONE;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+
+public class TestKuduIntegrationDynamicFilter
+        extends AbstractTestQueryFramework
+{
+    private static final int LINEITEM_COUNT = 60175;
+
+    private TestingKuduServer kuduServer;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        kuduServer = new TestingKuduServer();
+        return createKuduQueryRunnerTpch(
+                kuduServer,
+                Optional.of(""),
+                ImmutableMap.of("dynamic_filtering_wait_timeout", "1h"),
+                ImmutableMap.of(
+                        "dynamic-filtering.small-broadcast.max-distinct-values-per-driver", "100",
+                        "dynamic-filtering.small-broadcast.range-row-limit-per-driver", "100",
+                        "dynamic-filtering.large-broadcast.max-distinct-values-per-driver", "100",
+                        "dynamic-filtering.large-broadcast.range-row-limit-per-driver", "100000"),
+                TpchTable.getTables());
+    }
+
+    @AfterClass(alwaysRun = true)
+    public final void destroy()
+    {
+        kuduServer.close();
+    }
+
+    @Test(timeOut = 10_000)
+    public void testIncompleteDynamicFilterTimeout()
```

Comment:
This test should be on top of KuduSplitManager, similar to TestBackgroundHiveSplitLoader#testIncompleteDynamicFilterTimeout.
You shouldn't have to expose `completedOrTimeout` to test this.

Reply:
Updated